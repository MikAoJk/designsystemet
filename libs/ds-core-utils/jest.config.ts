/* eslint-disable */
export default {
  displayName: 'ds-core-utils',
  preset: '../../jest.preset.js',
  transform: {
    '^.+\\.[tj]sx?$': ['babel-jest', { presets: ['@nrwl/react/babel'] }],
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx'],
  coverageDirectory: '../../coverage/libs/ds-core-utils',
};
