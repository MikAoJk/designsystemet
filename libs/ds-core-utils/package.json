{
  "name": "@skatteetaten/ds-core-utils",
  "groupId": "no.skatteetaten.aurora",
  "version": "0.0.1",
  "publishConfig": {
    "registry": "https://nexus.sits.no/repository/npm-internal/"
  },
  "peerDependencies": {
    "react": ">=17"
  }
}
