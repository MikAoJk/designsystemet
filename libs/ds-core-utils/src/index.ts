export * from './SymbolLogo/SymbolLogo';
export * from './base-props.types';
export * from './defaults';
export * from './example-layouts/ExternalLayout';
export * from './example-layouts/InternalLayout';
export * from './i18n';
export * from './Separator/Separator';
