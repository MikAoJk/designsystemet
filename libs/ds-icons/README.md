# ds-icons

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test ds-icons` to execute the unit tests via [Jest](https://jestjs.io).
