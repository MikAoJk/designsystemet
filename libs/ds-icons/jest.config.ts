/* eslint-disable */
export default {
  displayName: 'ds-icons',
  preset: '../../jest.preset.js',
  transform: {
    '^.+\\.[tj]sx?$': ['babel-jest', { presets: ['@nx/react/babel'] }],
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx'],
  coverageDirectory: '../../coverage/libs/ds-icons',
};
