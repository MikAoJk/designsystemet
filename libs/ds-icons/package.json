{
  "name": "@skatteetaten/ds-icons",
  "groupId": "no.skatteetaten.aurora",
  "version": "0.5.0",
  "publishConfig": {
    "registry": "https://nexus.sits.no/repository/npm-internal/"
  },
  "peerDependencies": {
    "i18next": "^23",
    "react": "^18",
    "react-i18next": "^13",
    "@skatteetaten/ds-core-utils": "*"
  },
  "scripts": {}
}
