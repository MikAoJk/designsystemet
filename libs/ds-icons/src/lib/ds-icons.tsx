import styles from './ds-icons.module.scss';

export function DsIcons(): JSX.Element {
  return (
    <div className={styles['container']}>
      <h1>{'Welcome to DsIcons!'}</h1>
    </div>
  );
}

export default DsIcons;
