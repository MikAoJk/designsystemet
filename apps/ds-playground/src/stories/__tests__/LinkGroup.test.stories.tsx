import { AxePuppeteer } from '@axe-core/puppeteer';
import {
  LinkGroup,
  LinkGroupProps,
  linkGroupVariantArr,
  getLinkGroupVariantDefault,
} from '@skatteetaten/ds-buttons';
import { linkColorArr } from '@skatteetaten/ds-core-utils';
import { CalendarSVGpath } from '@skatteetaten/ds-icons';
import { ComponentMeta, ComponentStory } from '@storybook/react';
import { toHaveNoViolations } from 'jest-axe';
import { Page } from 'puppeteer';

import {
  screenShotOptions,
  wrapper,
} from './testUtils/puppeteer.testing.utils';

const verifyMatchSnapShot = async (page: Page): Promise<void> => {
  const innerHtml = await page.$eval(wrapper, (el) => el.innerHTML);
  expect(innerHtml).toMatchSnapshot();
};

const verifyMatchImageSnapShot = async (page: Page): Promise<void> => {
  const image = await page.screenshot(screenShotOptions);
  expect(image).toMatchImageSnapshot();
};

const verifyAxeRules = async (page: Page): Promise<void> => {
  const axeResults = await new AxePuppeteer(page).include(wrapper).analyze();
  expect.extend(toHaveNoViolations);
  expect(axeResults).toHaveNoViolations();
};

const verifySnapshotsAndAxeRules = async (page: Page): Promise<void> => {
  await verifyMatchSnapShot(page);
  await verifyMatchImageSnapShot(page);
  await verifyAxeRules(page);
};

export default {
  component: LinkGroup,
  title: 'Tester/LinkGroup',
  argTypes: {
    // Baseprops
    key: { table: { disable: true } },
    ref: { table: { disable: true } },
    className: { table: { disable: true } },
    id: { table: { disable: true } },
    lang: { table: { disable: true } },
    'data-testid': { table: { disable: true } },
    // Props
    children: { table: { disable: true } },
    hasSpacing: { table: { disable: true } },
    color: { table: { disable: true } },
    variant: {
      table: { disable: true },
      options: [...linkGroupVariantArr],
      control: 'inline-radio',
    },
  },
} as ComponentMeta<typeof LinkGroup>;

const Template: ComponentStory<typeof LinkGroup> = (args) => (
  <div className={'noTransition'} data-test-block>
    <LinkGroup {...args} />
  </div>
);

const defaultArgs: LinkGroupProps = {
  variant: getLinkGroupVariantDefault(),
  children: [
    <LinkGroup.Link key={'linkGroupLink_1'} href={'#root'}>
      {'Er du pendler?'}
    </LinkGroup.Link>,
    <LinkGroup.Link key={'linkGroupLink_2'} href={'#root'}>
      {'Pendler du mye?'}
    </LinkGroup.Link>,
    <LinkGroup.Link key={'linkGroupLink_3'} href={'#root'}>
      {'Pendler du dagen lang?'}
    </LinkGroup.Link>,
  ],
};

// Når LinkGroup har en ref, så får dom a elementet ref forwarded
export const WithRef = Template.bind({});
WithRef.storyName = 'With Ref (FA1)';
WithRef.args = {
  ...defaultArgs,
  ref: (instance: HTMLUListElement | null): void => {
    if (instance) {
      instance.id = 'dummyIdForwardedFromRef';
    }
  },
};
WithRef.argTypes = {
  ...WithRef.argTypes,
  ref: { table: { disable: false } },
};
WithRef.parameters = {
  async puppeteerTest(page: Page): Promise<void> {
    await verifyMatchSnapShot(page);
    await verifyAxeRules(page);

    const refId = await page.$eval(`${wrapper} > ul`, (el) => el.id);
    expect(refId).toBe('dummyIdForwardedFromRef');
  },
};

// Når LinkGroup har en id, så har ul-element id
export const WithId = Template.bind({});
WithId.storyName = 'With Id (FA2)';
WithId.args = {
  ...defaultArgs,
  id: 'htmlId',
};
WithId.argTypes = {
  ...WithId.argTypes,
  id: { table: { disable: false } },
};
WithId.parameters = {
  async puppeteerTest(page: Page): Promise<void> {
    await verifyMatchSnapShot(page);
    await verifyAxeRules(page);

    const id = await page.$eval(`${wrapper} > ul`, (el) =>
      el.getAttribute('id')
    );
    expect(id).toBe('htmlId');
  },
};

// Når LinkGroup har en custom CSS, så vises custom stil
export const WithCustomCss = Template.bind({});
WithCustomCss.storyName = 'With Custom CSS (FA3)';
WithCustomCss.args = {
  ...defaultArgs,
  className: 'dummyClassname',
};
WithCustomCss.argTypes = {
  ...WithCustomCss.argTypes,
  className: { table: { disable: false } },
};
WithCustomCss.parameters = {
  a11y: { disable: true },
  async puppeteerTest(page: Page): Promise<void> {
    // no axe rules because elements dont have sufficient color contrast
    await verifyMatchSnapShot(page);
    await verifyMatchImageSnapShot(page);

    const classNameAttribute = await page.$eval(`${wrapper}> ul`, (el) =>
      el.getAttribute('class')
    );
    expect(classNameAttribute).toContain('dummyClassname');
  },
};

// Når LinkGroup har en lang, så har ul-element lang
export const WithLang = Template.bind({});
WithLang.storyName = 'With Lang (FA4)';
WithLang.args = {
  ...defaultArgs,
  lang: 'nb',
};
WithLang.argTypes = {
  ...WithLang.argTypes,
  lang: { table: { disable: false } },
};
WithLang.parameters = {
  async puppeteerTest(page: Page): Promise<void> {
    await verifyMatchSnapShot(page);
    await verifyAxeRules(page);

    const langAttribute = await page.$eval(`${wrapper} > ul`, (el) =>
      el.getAttribute('lang')
    );
    expect(langAttribute).toBe('nb');
  },
};

// Når LinkGroup har dataTestid, så har ul-elementet data-testid satt
export const WithDataTestid = Template.bind({});
WithDataTestid.storyName = 'With DataTestid (FA5)';
WithDataTestid.args = {
  ...defaultArgs,
  'data-testid': '123ID',
};
WithDataTestid.argTypes = {
  ...WithDataTestid.argTypes,
  'data-testid': { table: { disable: false } },
};
WithDataTestid.parameters = {
  async puppeteerTest(page: Page): Promise<void> {
    await verifyMatchSnapShot(page);
    await verifyAxeRules(page);

    const dataTestid = await page.$eval(`${wrapper} > ul`, (el) =>
      el.getAttribute('data-testid')
    );
    expect(dataTestid).toBe('123ID');
  },
};

// Når LinkGroup instansieres, så vises default-variant list og rendrer riktig ulike tilstander (imageSnapshot viser kun ulike tilstander av listItem og ikke link)
export const Defaults = Template.bind({});
Defaults.storyName = 'Defaults Variant List (A1 delvis, A4 delvis, A5)';
Defaults.args = {
  ...defaultArgs,
};
Defaults.argTypes = {
  ...Defaults.argTypes,
  children: { table: { disable: false } },
};
Defaults.parameters = {
  async puppeteerTest(page: Page): Promise<void> {
    await verifySnapshotsAndAxeRules(page);

    const liElement = await page.$(`${wrapper} > ul > li:first-child`);

    await liElement?.hover();
    const imageHovered = await page.screenshot(screenShotOptions);
    expect(imageHovered).toMatchImageSnapshot();

    // active
    const cdp = await page.target().createCDPSession();
    const docNodeId = (await cdp.send('DOM.getDocument')).root.nodeId;
    const nodeId = (
      await cdp.send('DOM.querySelector', {
        nodeId: docNodeId,
        selector: `${wrapper} > ul > li:first-child`,
      })
    ).nodeId;

    await cdp.send('CSS.enable');
    await cdp.send('CSS.forcePseudoState', {
      nodeId: nodeId,
      forcedPseudoClasses: ['active'],
    });

    const imageActive = await page.screenshot(screenShotOptions);
    expect(imageActive).toMatchImageSnapshot();
  },
};

// Når LinkGroup instansieres, så vises variant anchors og rendrer riktig ulike tilstander (imageSnapshot viser kun ulike tilstander av listItem og ikke link)
export const VariantAnchors = Template.bind({});
VariantAnchors.storyName = 'Variant Anchors (A1 delvis, A4 delvis, A6)';
VariantAnchors.args = {
  ...defaultArgs,
  variant: 'anchors',
};
VariantAnchors.argTypes = {
  ...VariantAnchors.argTypes,
  variant: {
    table: {
      disable: false,
    },
  },
};
VariantAnchors.parameters = {
  async puppeteerTest(page: Page): Promise<void> {
    await verifySnapshotsAndAxeRules(page);

    const liElement = await page.$(`${wrapper} > ul > li:first-child`);

    await liElement?.hover();
    const imageHovered = await page.screenshot(screenShotOptions);
    expect(imageHovered).toMatchImageSnapshot();

    // active
    const cdp = await page.target().createCDPSession();
    const docNodeId = (await cdp.send('DOM.getDocument')).root.nodeId;
    const nodeId = (
      await cdp.send('DOM.querySelector', {
        nodeId: docNodeId,
        selector: `${wrapper} > ul > li:first-child`,
      })
    ).nodeId;

    await cdp.send('CSS.enable');
    await cdp.send('CSS.forcePseudoState', {
      nodeId: nodeId,
      forcedPseudoClasses: ['active'],
    });

    const imageActive = await page.screenshot(screenShotOptions);
    expect(imageActive).toMatchImageSnapshot();
  },
};

// Når LinkGroup har en veldig lang tekst, valgfritt ikon og/eller external ikon så skal teksten og ikon ha hengende innrykk
export const WithLongTextAndIcons = Template.bind({});
WithLongTextAndIcons.storyName = 'With Long Text And Icons (A2)';
WithLongTextAndIcons.args = {
  ...defaultArgs,
  children: [
    <LinkGroup.Link key={'linkGroupLink_1'} href={'#root'} isExternal>
      {'Denne lenken har en veldig lang tekst. Så lang at den lange teksten tvinger fram linjeskift med tekst som alltid er venstrejustert ' +
        'uansett om ikon eller ikke og får et hengende innrykk.'}
    </LinkGroup.Link>,
    <LinkGroup.Link
      key={'linkGroupLink_2'}
      href={'#root'}
      svgPath={CalendarSVGpath}
    >
      {'Denne lenken har en veldig lang tekst. Så lang at den lange teksten tvinger fram linjeskift med tekst som alltid er venstrejustert ' +
        'uansett om ikon eller ikke og får et hengende innrykk.'}
    </LinkGroup.Link>,
  ],
};
WithLongTextAndIcons.argTypes = {
  ...WithLongTextAndIcons.argTypes,
  children: { table: { disable: false } },
};
WithLongTextAndIcons.parameters = {
  puppeteerTest: verifySnapshotsAndAxeRules,
};

const TemplateWithTwoLinkGroups: ComponentStory<typeof LinkGroup> = (args) => (
  <div data-test-block>
    <LinkGroup {...args} />
    <LinkGroup {...args} />
  </div>
);

// Når LinkGroup har spacing, så får gruppen en margin under gruppen
export const WithSpacing = TemplateWithTwoLinkGroups.bind({});
WithSpacing.storyName = 'With Spacing (A3)';
WithSpacing.args = {
  ...defaultArgs,
  hasSpacing: true,
};
WithSpacing.argTypes = {
  ...WithSpacing.argTypes,
  hasSpacing: { table: { disable: false } },
};
WithSpacing.parameters = {
  puppeteerTest: verifySnapshotsAndAxeRules,
};

// Når LinkGroup har color white, så vises tekster og ikoner i hvit
export const WithColor = Template.bind({});
WithColor.storyName = 'With Color (A7)';
WithColor.args = {
  ...defaultArgs,
  color: 'white',
};
WithColor.argTypes = {
  ...WithColor.argTypes,
  color: {
    options: ['default', ...linkColorArr],
    mapping: {
      default: '',
      ...linkColorArr,
    },
    table: { disable: false },
  },
};
WithColor.parameters = {
  backgrounds: {
    default: 'themePrimary',
  },
  async puppeteerTest(page: Page): Promise<void> {
    await verifyMatchImageSnapShot(page);
    await verifyAxeRules(page);
  },
};
