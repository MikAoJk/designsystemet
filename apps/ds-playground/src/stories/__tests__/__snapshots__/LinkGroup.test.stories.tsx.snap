// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tester/LinkGroup Defaults smoke-test 1`] = `
<div data-test-block="true">
  <ul class="LinkGroup_linkGroup__ZiEgs   ">
    <li class="LinkGroup_linkGroupItem__Y7SjF">
      <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz LinkGroup_icon__cSIB- LinkGroup_icon_forward__ErPTi "
           viewbox="0 0 24 24"
           focusable="false"
           aria-hidden="true"
           role="img"
      >
        <path d="M1.8 12.6v-1.8h16.8l-5.2-5.1 1.4-1.3 7.5 7.3-7.7 7.3-1.4-1.3 5.2-5.1H1.8Z">
        </path>
      </svg>
      <a class="Link_link__jeUpY  "
         href="#storybook-root"
      >
        Er du pendler?
      </a>
    </li>
    <li class="LinkGroup_linkGroupItem__Y7SjF">
      <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz LinkGroup_icon__cSIB- LinkGroup_icon_forward__ErPTi "
           viewbox="0 0 24 24"
           focusable="false"
           aria-hidden="true"
           role="img"
      >
        <path d="M1.8 12.6v-1.8h16.8l-5.2-5.1 1.4-1.3 7.5 7.3-7.7 7.3-1.4-1.3 5.2-5.1H1.8Z">
        </path>
      </svg>
      <a class="Link_link__jeUpY  "
         href="#storybook-root"
      >
        Pendler du mye?
      </a>
    </li>
    <li class="LinkGroup_linkGroupItem__Y7SjF">
      <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz LinkGroup_icon__cSIB- LinkGroup_icon_forward__ErPTi "
           viewbox="0 0 24 24"
           focusable="false"
           aria-hidden="true"
           role="img"
      >
        <path d="M1.8 12.6v-1.8h16.8l-5.2-5.1 1.4-1.3 7.5 7.3-7.7 7.3-1.4-1.3 5.2-5.1H1.8Z">
        </path>
      </svg>
      <a class="Link_link__jeUpY  "
         href="#storybook-root"
      >
        Pendler du dagen lang?
      </a>
    </li>
  </ul>
</div>
`;

exports[`Tester/LinkGroup VariantAnchors smoke-test 1`] = `
<div data-test-block="true">
  <ul class="LinkGroup_linkGroup__ZiEgs   ">
    <li class="LinkGroup_linkGroupItem__Y7SjF">
      <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz LinkGroup_icon__cSIB- LinkGroup_icon_down__oXF-n "
           viewbox="0 0 24 24"
           focusable="false"
           aria-hidden="true"
           role="img"
      >
        <path d="M11.1 1.5h1.8v16.8l5.1-5.2 1.3 1.4L12 22l-7.3-7.6L6 13l5.1 5.2V1.5Z">
        </path>
      </svg>
      <a class="Link_link__jeUpY  "
         href="#storybook-root"
      >
        Er du pendler?
      </a>
    </li>
    <li class="LinkGroup_linkGroupItem__Y7SjF">
      <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz LinkGroup_icon__cSIB- LinkGroup_icon_down__oXF-n "
           viewbox="0 0 24 24"
           focusable="false"
           aria-hidden="true"
           role="img"
      >
        <path d="M11.1 1.5h1.8v16.8l5.1-5.2 1.3 1.4L12 22l-7.3-7.6L6 13l5.1 5.2V1.5Z">
        </path>
      </svg>
      <a class="Link_link__jeUpY  "
         href="#storybook-root"
      >
        Pendler du mye?
      </a>
    </li>
    <li class="LinkGroup_linkGroupItem__Y7SjF">
      <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz LinkGroup_icon__cSIB- LinkGroup_icon_down__oXF-n "
           viewbox="0 0 24 24"
           focusable="false"
           aria-hidden="true"
           role="img"
      >
        <path d="M11.1 1.5h1.8v16.8l5.1-5.2 1.3 1.4L12 22l-7.3-7.6L6 13l5.1 5.2V1.5Z">
        </path>
      </svg>
      <a class="Link_link__jeUpY  "
         href="#storybook-root"
      >
        Pendler du dagen lang?
      </a>
    </li>
  </ul>
</div>
`;

exports[`Tester/LinkGroup WithAttributes play-test 1`] = `
<div data-test-block="true">
  <ul id="htmlId"
      class="LinkGroup_linkGroup__ZiEgs  dummyClassname "
      lang="nb"
      data-testid="123ID"
  >
    <li class="LinkGroup_linkGroupItem__Y7SjF">
      <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz LinkGroup_icon__cSIB- LinkGroup_icon_forward__ErPTi "
           viewbox="0 0 24 24"
           focusable="false"
           aria-hidden="true"
           role="img"
      >
        <path d="M1.8 12.6v-1.8h16.8l-5.2-5.1 1.4-1.3 7.5 7.3-7.7 7.3-1.4-1.3 5.2-5.1H1.8Z">
        </path>
      </svg>
      <a class="Link_link__jeUpY  "
         href="#storybook-root"
      >
        Er du pendler?
      </a>
    </li>
    <li class="LinkGroup_linkGroupItem__Y7SjF">
      <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz LinkGroup_icon__cSIB- LinkGroup_icon_forward__ErPTi "
           viewbox="0 0 24 24"
           focusable="false"
           aria-hidden="true"
           role="img"
      >
        <path d="M1.8 12.6v-1.8h16.8l-5.2-5.1 1.4-1.3 7.5 7.3-7.7 7.3-1.4-1.3 5.2-5.1H1.8Z">
        </path>
      </svg>
      <a class="Link_link__jeUpY  "
         href="#storybook-root"
      >
        Pendler du mye?
      </a>
    </li>
    <li class="LinkGroup_linkGroupItem__Y7SjF">
      <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz LinkGroup_icon__cSIB- LinkGroup_icon_forward__ErPTi "
           viewbox="0 0 24 24"
           focusable="false"
           aria-hidden="true"
           role="img"
      >
        <path d="M1.8 12.6v-1.8h16.8l-5.2-5.1 1.4-1.3 7.5 7.3-7.7 7.3-1.4-1.3 5.2-5.1H1.8Z">
        </path>
      </svg>
      <a class="Link_link__jeUpY  "
         href="#storybook-root"
      >
        Pendler du dagen lang?
      </a>
    </li>
  </ul>
</div>
`;

exports[`Tester/LinkGroup WithColor smoke-test 1`] = `
<div data-test-block="true">
  <ul class="LinkGroup_linkGroup__ZiEgs   ">
    <li class="LinkGroup_linkGroupItem__Y7SjF">
      <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz LinkGroup_icon__cSIB- LinkGroup_icon_forward__ErPTi LinkGroup_icon_white__5ZXue"
           viewbox="0 0 24 24"
           focusable="false"
           aria-hidden="true"
           role="img"
      >
        <path d="M1.8 12.6v-1.8h16.8l-5.2-5.1 1.4-1.3 7.5 7.3-7.7 7.3-1.4-1.3 5.2-5.1H1.8Z">
        </path>
      </svg>
      <a class="Link_link__jeUpY Link_link_white__bmeQu "
         href="#storybook-root"
      >
        Er du pendler?
      </a>
    </li>
    <li class="LinkGroup_linkGroupItem__Y7SjF">
      <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz LinkGroup_icon__cSIB- LinkGroup_icon_forward__ErPTi LinkGroup_icon_white__5ZXue"
           viewbox="0 0 24 24"
           focusable="false"
           aria-hidden="true"
           role="img"
      >
        <path d="M1.8 12.6v-1.8h16.8l-5.2-5.1 1.4-1.3 7.5 7.3-7.7 7.3-1.4-1.3 5.2-5.1H1.8Z">
        </path>
      </svg>
      <a class="Link_link__jeUpY Link_link_white__bmeQu "
         href="#storybook-root"
      >
        Pendler du mye?
      </a>
    </li>
    <li class="LinkGroup_linkGroupItem__Y7SjF">
      <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz LinkGroup_icon__cSIB- LinkGroup_icon_forward__ErPTi LinkGroup_icon_white__5ZXue"
           viewbox="0 0 24 24"
           focusable="false"
           aria-hidden="true"
           role="img"
      >
        <path d="M1.8 12.6v-1.8h16.8l-5.2-5.1 1.4-1.3 7.5 7.3-7.7 7.3-1.4-1.3 5.2-5.1H1.8Z">
        </path>
      </svg>
      <a class="Link_link__jeUpY Link_link_white__bmeQu "
         href="#storybook-root"
      >
        Pendler du dagen lang?
      </a>
    </li>
  </ul>
</div>
`;

exports[`Tester/LinkGroup WithLongTextAndIcons smoke-test 1`] = `
<div data-test-block="true">
  <ul class="LinkGroup_linkGroup__ZiEgs   ">
    <li class="LinkGroup_linkGroupItem__Y7SjF">
      <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz LinkGroup_icon__cSIB- LinkGroup_icon_forward__ErPTi "
           viewbox="0 0 24 24"
           focusable="false"
           aria-hidden="true"
           role="img"
      >
        <path d="M1.8 12.6v-1.8h16.8l-5.2-5.1 1.4-1.3 7.5 7.3-7.7 7.3-1.4-1.3 5.2-5.1H1.8Z">
        </path>
      </svg>
      <a class="Link_link__jeUpY  "
         href="#storybook-root"
      >
        Denne lenken har en veldig lang tekst. S책 lang at den lange teksten tvinger fram linjeskift med tekst som alltid er venstrejustert uansett om ikon eller ikke og f책r et hengende innrykk.
        <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_small__Yzxmo Link_icon__iy55p Link_icon_isExternal__O+HIA"
             viewbox="0 0 24 24"
             focusable="false"
             aria-label="Til et annet nettsted"
             aria-hidden="false"
             role="img"
        >
          <path d="M14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7Zm5 16H5V5h7V3H5a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2v-7h-2v7Z">
          </path>
        </svg>
      </a>
    </li>
    <li class="LinkGroup_linkGroupItem__Y7SjF">
      <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz LinkGroup_icon__cSIB- LinkGroup_icon_forward__ErPTi "
           viewbox="0 0 24 24"
           focusable="false"
           aria-hidden="true"
           role="img"
      >
        <path d="M1.8 12.6v-1.8h16.8l-5.2-5.1 1.4-1.3 7.5 7.3-7.7 7.3-1.4-1.3 5.2-5.1H1.8Z">
        </path>
      </svg>
      <a class="Link_link__jeUpY  "
         href="#storybook-root"
      >
        <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_small__Yzxmo Link_icon__iy55p Link_icon_isCustom__nsOot"
             viewbox="0 0 24 24"
             focusable="false"
             aria-hidden="true"
             role="img"
        >
          <path d="M19 19H5V8h14v11ZM16 1v2H8V1H6v2H5c-1.11 0-2 .89-2 2v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2V5a2 2 0 0 0-2-2h-1V1h-2Zm1 11h-5v5h5v-5Z">
          </path>
        </svg>
        Denne lenken har en veldig lang tekst. S책 lang at den lange teksten tvinger fram linjeskift med tekst som alltid er venstrejustert uansett om ikon eller ikke og f책r et hengende innrykk.
      </a>
    </li>
  </ul>
</div>
`;

exports[`Tester/LinkGroup WithRef play-test 1`] = `
<div data-test-block="true">
  <ul class="LinkGroup_linkGroup__ZiEgs   "
      id="dummyIdForwardedFromRef"
  >
    <li class="LinkGroup_linkGroupItem__Y7SjF">
      <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz LinkGroup_icon__cSIB- LinkGroup_icon_forward__ErPTi "
           viewbox="0 0 24 24"
           focusable="false"
           aria-hidden="true"
           role="img"
      >
        <path d="M1.8 12.6v-1.8h16.8l-5.2-5.1 1.4-1.3 7.5 7.3-7.7 7.3-1.4-1.3 5.2-5.1H1.8Z">
        </path>
      </svg>
      <a class="Link_link__jeUpY  "
         href="#storybook-root"
      >
        Er du pendler?
      </a>
    </li>
    <li class="LinkGroup_linkGroupItem__Y7SjF">
      <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz LinkGroup_icon__cSIB- LinkGroup_icon_forward__ErPTi "
           viewbox="0 0 24 24"
           focusable="false"
           aria-hidden="true"
           role="img"
      >
        <path d="M1.8 12.6v-1.8h16.8l-5.2-5.1 1.4-1.3 7.5 7.3-7.7 7.3-1.4-1.3 5.2-5.1H1.8Z">
        </path>
      </svg>
      <a class="Link_link__jeUpY  "
         href="#storybook-root"
      >
        Pendler du mye?
      </a>
    </li>
    <li class="LinkGroup_linkGroupItem__Y7SjF">
      <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz LinkGroup_icon__cSIB- LinkGroup_icon_forward__ErPTi "
           viewbox="0 0 24 24"
           focusable="false"
           aria-hidden="true"
           role="img"
      >
        <path d="M1.8 12.6v-1.8h16.8l-5.2-5.1 1.4-1.3 7.5 7.3-7.7 7.3-1.4-1.3 5.2-5.1H1.8Z">
        </path>
      </svg>
      <a class="Link_link__jeUpY  "
         href="#storybook-root"
      >
        Pendler du dagen lang?
      </a>
    </li>
  </ul>
</div>
`;

exports[`Tester/LinkGroup WithSpacing smoke-test 1`] = `
<div data-test-block="true">
  <ul class="LinkGroup_linkGroup__ZiEgs LinkGroup_linkGroup_hasSpacing__AhexL  ">
    <li class="LinkGroup_linkGroupItem__Y7SjF">
      <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz LinkGroup_icon__cSIB- LinkGroup_icon_forward__ErPTi "
           viewbox="0 0 24 24"
           focusable="false"
           aria-hidden="true"
           role="img"
      >
        <path d="M1.8 12.6v-1.8h16.8l-5.2-5.1 1.4-1.3 7.5 7.3-7.7 7.3-1.4-1.3 5.2-5.1H1.8Z">
        </path>
      </svg>
      <a class="Link_link__jeUpY  "
         href="#storybook-root"
      >
        Er du pendler?
      </a>
    </li>
    <li class="LinkGroup_linkGroupItem__Y7SjF">
      <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz LinkGroup_icon__cSIB- LinkGroup_icon_forward__ErPTi "
           viewbox="0 0 24 24"
           focusable="false"
           aria-hidden="true"
           role="img"
      >
        <path d="M1.8 12.6v-1.8h16.8l-5.2-5.1 1.4-1.3 7.5 7.3-7.7 7.3-1.4-1.3 5.2-5.1H1.8Z">
        </path>
      </svg>
      <a class="Link_link__jeUpY  "
         href="#storybook-root"
      >
        Pendler du mye?
      </a>
    </li>
    <li class="LinkGroup_linkGroupItem__Y7SjF">
      <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz LinkGroup_icon__cSIB- LinkGroup_icon_forward__ErPTi "
           viewbox="0 0 24 24"
           focusable="false"
           aria-hidden="true"
           role="img"
      >
        <path d="M1.8 12.6v-1.8h16.8l-5.2-5.1 1.4-1.3 7.5 7.3-7.7 7.3-1.4-1.3 5.2-5.1H1.8Z">
        </path>
      </svg>
      <a class="Link_link__jeUpY  "
         href="#storybook-root"
      >
        Pendler du dagen lang?
      </a>
    </li>
  </ul>
  <ul class="LinkGroup_linkGroup__ZiEgs LinkGroup_linkGroup_hasSpacing__AhexL  ">
    <li class="LinkGroup_linkGroupItem__Y7SjF">
      <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz LinkGroup_icon__cSIB- LinkGroup_icon_forward__ErPTi "
           viewbox="0 0 24 24"
           focusable="false"
           aria-hidden="true"
           role="img"
      >
        <path d="M1.8 12.6v-1.8h16.8l-5.2-5.1 1.4-1.3 7.5 7.3-7.7 7.3-1.4-1.3 5.2-5.1H1.8Z">
        </path>
      </svg>
      <a class="Link_link__jeUpY  "
         href="#storybook-root"
      >
        Er du pendler?
      </a>
    </li>
    <li class="LinkGroup_linkGroupItem__Y7SjF">
      <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz LinkGroup_icon__cSIB- LinkGroup_icon_forward__ErPTi "
           viewbox="0 0 24 24"
           focusable="false"
           aria-hidden="true"
           role="img"
      >
        <path d="M1.8 12.6v-1.8h16.8l-5.2-5.1 1.4-1.3 7.5 7.3-7.7 7.3-1.4-1.3 5.2-5.1H1.8Z">
        </path>
      </svg>
      <a class="Link_link__jeUpY  "
         href="#storybook-root"
      >
        Pendler du mye?
      </a>
    </li>
    <li class="LinkGroup_linkGroupItem__Y7SjF">
      <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz LinkGroup_icon__cSIB- LinkGroup_icon_forward__ErPTi "
           viewbox="0 0 24 24"
           focusable="false"
           aria-hidden="true"
           role="img"
      >
        <path d="M1.8 12.6v-1.8h16.8l-5.2-5.1 1.4-1.3 7.5 7.3-7.7 7.3-1.4-1.3 5.2-5.1H1.8Z">
        </path>
      </svg>
      <a class="Link_link__jeUpY  "
         href="#storybook-root"
      >
        Pendler du dagen lang?
      </a>
    </li>
  </ul>
</div>
`;
